tokio/src/runtime/basic_scheduler.rs

```diff
@@ -482,6 +494,8 @@ impl Schedule for SchedulerPriv {
 
     fn release_local(&self, task: &Task<Self>) {
         unsafe {
+            // safety: Scheduler::release_local is only called from the
+            // scheduler's thread.
```

Comment:
How are we guaranteed to be on the same thread? Is `SchedulerPriv` not `Sync`/`Send`?

Reply:
The `release_local` method is only _called_ from the same thread as the scheduler — it's part of the `Schedule` trait's contract.

Reply:
Ah, okay, then it'd be good to include that in the comment.

Reply:
reworded in https://github.com/tokio-rs/tokio/pull/1892/commits/03357f1b78744fd4bf3f963859f8573c22b25d6e